corehq/ex-submodules/couchexport/writers.py

```diff
@@ -581,32 +584,98 @@ def _find_geo_property_by_path(table):
+            if not coordinates.get('lng') or not coordinates.get('lat'):
                 continue
-            properties = {header: row[i] for i, header in enumerate(table_headers) if header != geo_property_name}
+
             features.append(self.parse_feature(
-                coordinates=[lng, lat],
+                coordinates=[coordinates['lng'], coordinates['lat']],
                 properties=properties,
             ))
+
         return features
 
+    def find_geo_property_header(self, table, table_headers):
+        selected_geo_property_header = table.selected_geo_property
+        if selected_geo_property_header not in table_headers:
+            selected_geo_property_header = self._find_geo_property_by_path(table)
+        return selected_geo_property_header
+
+    @staticmethod
+    def find_geo_data_column(headers, geo_property_name):
+        if geo_property_name not in headers:
+            return None
+        return headers.index(geo_property_name)
+
+    @staticmethod
+    def find_geo_data_columns(headers, geo_property_name):
+        """
+        Finds the indices of the latitude and longitude columns relating to the geo_column_name
+        and returns the indices in that specific order, i.e. <lat_index>, <lng_index>
+        """
+        longitude_column = GPS_SPLIT_COLUMN_LONGITUDE_TEMPLATE.format(geo_property_name)
+        latitude_column = GPS_SPLIT_COLUMN_LATITUDE_TEMPLATE.format(geo_property_name)
+
+        if longitude_column not in headers or latitude_column not in headers:
+            return []
+
+        return [headers.index(latitude_column), headers.index(longitude_column)]
+
+    @staticmethod
+    def collect_geo_data(table_headers, row, col_index):
+        coordinate_data = row[col_index].split(" ")
+        # check for invalid data in row
+        if len(coordinate_data) < 2:
+            return {}, {}
+
+        lat, lng = coordinate_data[0], coordinate_data[1]
+
+        properties = {
+            header: row[i]
+            for i, header in enumerate(table_headers)
+            if i != col_index
+        }
+        return {'lat': lat, 'lng': lng}, properties
+
+    @staticmethod
+    def collect_geo_data_multi_column(table_headers, row, col_index):
+        lat = row[col_index[0]]
+        lng = row[col_index[1]]
+
+        properties = {
+            header: row[i]
+            for i, header in enumerate(table_headers)
+            if i not in col_index
+        }
+        return {'lat': lat, 'lng': lng}, properties
+
+    def get_features(self, table, data):
+        table_headers = data[0]
+        geo_column_index_func = self.find_geo_data_column
+        geo_data_collector_func = self.collect_geo_data
+
+        if getattr(table, 'split_multiselects', False):
+            geo_column_index_func = self.find_geo_data_columns
+            geo_data_collector_func = self.collect_geo_data_multi_column
```

Comment:
nit:
its totally up to preference. I'd do this one as
```
if getattr(table, 'split_multiselects', False):
    geo_column_index_func = self.find_geo_data_columns
    geo_data_collector_func = self.collect_geo_data_multi_column
else:
    geo_column_index_func = self.find_geo_data_column
    geo_data_collector_func = self.collect_geo_data
```

to make the decision making here more explicit.